.github/workflows/deploy_workflow.yml

```diff
@@ -0,0 +1,252 @@
+on:
+    workflow_call:
+      inputs:
+        build_prefix:
+          type: string
+        features:
+          type: string
+          default: "testing"
+      secrets:
+        account_id:
+          required: true
+        instance_id:
+          required: true
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-1
+          role-to-assume: arn:aws:iam::${{ secrets.account_id }}:role/midendev-GithubActionsRole
+          role-session-name: GithubActionsSession
+
+      - name: Stop services
+        id: stop_services
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.instance_id }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":["sudo systemctl stop miden-node", "sudo systemctl stop miden-faucet"]}' \
+            --output text \
+            --query "Command.CommandId")
+          echo "command_id=$COMMAND_ID" >> $GITHUB_OUTPUT
+
+      - name: Check stop services command status and retrieve output
+        run: |
+          while true; do
+            STATUS=$(aws ssm list-command-invocations \
+                --command-id ${{ steps.stop_services.outputs.command_id }} \
+                --details \
+                --query "CommandInvocations[0].Status" \
+                --output text)
+            echo "Command Status: $STATUS"
+            
+            OUTPUT=$(aws ssm list-command-invocations \
+                --command-id ${{ steps.stop_services.outputs.command_id }} \
+                --details \
+                --query "CommandInvocations[0].CommandPlugins[0].Output" \
+                --output text)
+            echo "Command Output: $OUTPUT"
+            
+            if [ "$STATUS" == "Success" ]; then
+              echo "Command completed successfully."
+              break
+            elif [ "$STATUS" == "Failed" ] || [ "$STATUS" == "Cancelled" ]; then
+              echo "Command failed with status: $STATUS"
+              exit 1
+            else
+              sleep 10  # Wait for 30 seconds before checking again
+            fi
+          done
+      
+      - name: Install prerequisites
+        id: install_prereq
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.instance_id }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":[
+                  "sudo apt-get update -y",
+                  "sudo apt-get install -y git gcc openssl libbz2-dev libffi7 make",
+                  "sudo apt install -y build-essential",
+                  "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y",
+                  "source $HOME/.cargo/env",
```

Comment:
Is this required because the rust install script doesn't update the current sessions paths?